pkg/url/url.go

```diff
@@ -100,6 +104,74 @@ func GetIngress(kClient *kclient.Client, envSpecificInfo *envinfo.EnvSpecificInf
 	return iextensionsv1.Ingress{}, errors.New(fmt.Sprintf("the url %v does not exist", urlName))
 }
 
+// GetContainer returns Docker URL definition for given URL name
+func GetContainerURL(client *lclient.Client, envSpecificInfo *envinfo.EnvSpecificInfo, urlName string, componentName string) (URL, error) {
+	localURLs := envSpecificInfo.GetURL()
+	containers, err := dockerutils.GetComponentContainers(*client, componentName)
+	if err != nil {
+		return URL{}, errors.Wrap(err, "unable to get component containers")
+	}
+	containerJSONMap := make(map[string]types.ContainerJSON)
+	for _, c := range containers {
+		containerJSON, err := client.Client.ContainerInspect(client.Context, c.ID)
+		if err != nil {
+			return URL{}, err
+		}
+		containerJSONMap[c.ID] = containerJSON
+		for internalPort, portbinding := range containerJSON.HostConfig.PortBindings {
```

Comment:
Comments on this part? Why are we going through each range, what happens each loop?

Reply:
updated